core/src/main/java/org/bitcoinj/core/TransactionInput.java

```diff
@@ -549,4 +555,38 @@ public String toString() {
             throw new RuntimeException(e);
         }
     }
+
+    /**
+     * Gets the sum of a collection of inputs as a {@link Coin}
+     * <p>
+     * Note: inputs can have a null Coin value. Those inputs will be skipped and a warning logged.
+     * @param inputs collection of transaction inputs
+     * @return the sum
+     * @throws ArithmeticException if overflows occurred in signed {@code long} arithmetic
+     */
+    public static Coin sum(Collection<TransactionInput> inputs) {
+        return Coin.ofSat(sumSats(inputs));
+    }
+
+    /**
+     * Gets the sum of a collection of inputs in satoshis
+     * <p>
+     * Note: inputs can have a null Coin value. Those inputs will be skipped and a warning logged.
+     * TODO: See related TODO in Wallet.completeTx
+     * @param inputs collection of transaction inputs
+     * @return the sum (in satoshis)
+     * @throws ArithmeticException if overflows occurred in signed {@code long} arithmetic
+     */
+    public static long sumSats(Collection<TransactionInput> inputs) {
+        return inputs.stream()
+                .peek(input -> {
+                    if (input.value == null) {
+                        log.warn("input with unknown value: {}", input);
+                    }
+                })
```

Comment:
I put this logging here after seeing the logging in Wallet.java on line 4192.

Reply:
I assume you mean the
`log.warn("SendRequest transaction already has inputs but we don't know how much they are worth - they will be added to fee.");`
line? So what kind of comment are you asking for here?

Reply:
The message explains that if you pick the output but don't know its value (and assume zero), you'll implicitly add its value to the fee (since you are sending less to its outputs). This might lead in an unexpected high fee and thus loss of Bitcoins.

However, I think since SegWit this situation is quite unlikely as the input value is part of the sigHash. So you need to cryptographically prove to know it.

For me, the question becomes: why don't we throw an exception at this point? I think this is because the SendRequest API somehow supports passing in half-baked transactions received from elsewhere (e.g. multisig protocol) and let bitcoinj add the missing bits. However I don't understand how those usecases work to prevent the "loosing Bitcoins to fee" problem.